src/App.tsx

```diff
@@ -15,35 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+
+
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [reversePressed, setReversePressed] = useState(false);
+  const [alphabeticallyPressed, setAlphabetically] = useState(false);
+  const [byLengthPressed, setByLength] = useState(false);
+
+  const sortGoods = (sortType: string) => {
+    let sorted = [...sortedGoods];
+
+    switch (sortType) {
+      case 'alphabetically':
+        sorted.sort((a, b) => {
+          setAlphabetically(!alphabeticallyPressed);
+          if (byLengthPressed) {
+            setByLength(!byLengthPressed);
+          }
+          if (reversePressed) {
+            return a.localeCompare(b) * -1;
+          } else {
+            return a.localeCompare(b)
+          }
+        })
```

Comment:
The state updates for `alphabeticallyPressed` and `byLengthPressed` should not be inside the `sort` function. State updates are asynchronous and may not reflect immediately, leading to potential bugs. Consider moving these updates outside the `sort` function.